client/allocrunner/taskrunner/getter/util_test.go

```diff
@@ -240,3 +241,33 @@ func TestUtil_environment(t *testing.T) {
 		}, result)
 	})
 }
+
+func TestUtil_isPathWithin(t *testing.T) {
+	tdir := t.TempDir()
+	t.Run("when path not within root", func(t *testing.T) {
+		root := mktmpdir(t, tdir)
+		check := mktmpdir(t, tdir)
+
+		must.False(t, isPathWithin(root, check))
+	})
+
+	t.Run("when path within root", func(t *testing.T) {
+		root := mktmpdir(t, tdir)
+		check := mktmpdir(t, root)
+
+		must.True(t, isPathWithin(root, check))
+	})
+
+	t.Run("when root within path", func(t *testing.T) {
+		check := mktmpdir(t, tdir)
+		root := mktmpdir(t, check)
+
+		must.False(t, isPathWithin(root, check))
+	})
+}
+
+func mktmpdir(t *testing.T, parent string) string {
```

Comment:
I believe you've reinvented [`t.TempDir`](https://pkg.go.dev/testing#T.TempDir) -- there are some other `os.MkdirTemp`s around, too, which could use `t.TempDir` instead.

Reply:
This is just a little helper for creating directories within a given parent so it can just be a single line call instead of having to do error checking on all of them. The name is misleading, though, and since it's only being used within one test I'll relocate it to make it a bit more clear.